pkg/jobs/hive/hive.go

```diff
@@ -646,3 +1004,113 @@ func retreiveAndUpdateClusterVersion(
+
+	// Get latest clusterversion or else the object might be stale
+	mcview := managedclusterviewv1beta1.ManagedClusterView{}
+	mcaStatus := managedclusteractionv1beta1.ManagedClusterAction{}
+	resultmcview := managedclusterviewv1beta1.ManagedClusterView{}
+	clusterVersion := map[string]interface{}{}
+
+	if err := client.Get(context.TODO(), types.NamespacedName{
+		Namespace: clusterName,
+		Name:      clusterName,
+	}, &mcview); err != nil {
+
+		klog.V(2).Info("Create managedclusterview " + clusterName)
+		if err := client.Create(context.TODO(), managedclusterview); err != nil {
+			return mcaStatus, err
+		}
+	}
+
+	if err := waitForMCV(client, clusterName, clusterName, &resultmcview, getErr); err != nil {
+		return mcaStatus, err
+	}
+	resultClusterVersion := resultmcview.Status.Result
+
+	if resultClusterVersion.Raw != nil {
+		err := json.Unmarshal(resultClusterVersion.Raw, &clusterVersion)
+		utils.CheckError(err)
+	} else {
+		return mcaStatus, getErr
+	}
+
+	cvDesiredUpdate := clusterVersion["spec"].(map[string]interface{})["desiredUpdate"]
+
+	// Always use 'force' option since we cannot get the image hash for EUS to EUS upgrade
+	// Using the version image tag to upgrade will be blocked
+	if cvDesiredUpdate != nil {
+		cvDesiredUpdate.(map[string]interface{})["version"] = updateVersion
+		cvDesiredUpdate.(map[string]interface{})["force"] = true
+		cvDesiredUpdate.(map[string]interface{})["image"] =
+			"quay.io/openshift-release-dev/ocp-release:" + updateVersion + "-multi"
+	} else {
+		// For when desiredUpdate does not exist
+		clusterVersion["spec"].(map[string]interface{})["desiredUpdate"] = map[string]interface{}{
+			"version": updateVersion,
+			"force":   true,
+			"image":   "quay.io/openshift-release-dev/ocp-release:" + updateVersion + "-multi",
+		}
+	}
+
+	// set channel when upgrading to final EUS version
+	if !isInterVersion {
+		finalSemVer, err := semver.Make(updateVersion)
+		if err != nil {
+			return mcaStatus, err
+		}
+		clusterVersion["spec"].(map[string]interface{})["channel"] = "stable-" + strconv.Itoa(int(finalSemVer.Major)) +
+			"." + strconv.Itoa(int(finalSemVer.Minor))
+	}
+
+	var updateClusterVersion runtime.RawExtension
+	if clusterVersion != nil {
+		b, err := json.Marshal(clusterVersion)
+		utils.CheckError(err)
+		updateClusterVersion.Raw = b
+	}
+	klog.V(2).Info("Create managedclusteraction to update clusterversion " + clusterName)
+	managedclusteraction := &managedclusteractionv1beta1.ManagedClusterAction{
+		ObjectMeta: v1.ObjectMeta{
+			Name:      clusterName,
+			Namespace: clusterName,
+		},
+		Spec: managedclusteractionv1beta1.ActionSpec{
+			ActionType: managedclusteractionv1beta1.UpdateActionType,
+			KubeWork: &managedclusteractionv1beta1.KubeWorkSpec{
+				Resource:       "clusterversions",
+				Name:           "version",
+				Namespace:      "",
+				ObjectTemplate: updateClusterVersion,
+			},
+		},
+	}
+	if err := client.Create(context.TODO(), managedclusteraction); err != nil {
```

Comment:
Does it need a Get to make sure it doesn't already exist before proceed with the Create?